src/extensionsIntegrated/CSSColorHints/main.js

```diff
@@ -0,0 +1,150 @@
+define(function (require, exports, module) {
+    // Brackets modules
+    const ColorUtils = require('utils/ColorUtils'),
+        AppInit = require("utils/AppInit"),
+        CSSUtils = require("language/CSSUtils"),
+        CodeHintManager = require("editor/CodeHintManager");
+
+    const COLOR_PROPERTIES = [
+        "color",
+        "background-color",
+        "border-color",
+        "border-top-color",
+        "border-right-color",
+        "border-bottom-color",
+        "border-left-color",
+        "outline-color",
+        "text-decoration-color",
+        "text-emphasis-color",
+        "text-shadow",
+        "box-shadow",
+        "background",
+        "border",
+        "border-top",
+        "border-right",
+        "border-bottom",
+        "border-left"
+    ];
+
+    const languages = ["css", "scss", "less", "sass", "stylus", "html", "svg", "jsx", "tsx"];
+
+    let editor = null;
+    let cursorInfo = null;
+
+    function _isAlphanumeric(char) {
+        return /^[a-z0-9-@$]$/i.test(char);
+    }
+
+    function _isColor(segment, colorInSegment, colorIndex) {
+        const previousChar = colorIndex === 0 ? "" : segment.charAt(colorIndex - 1);
+        const endIndex = colorIndex + colorInSegment.length;
+        const nextChar = endIndex === segment.length ? "" : segment.charAt(endIndex);
+        return !_isAlphanumeric(previousChar) && !_isAlphanumeric(nextChar);
+    }
+
+    function _isColorProperty(propertyName) {
+        return COLOR_PROPERTIES.includes(propertyName.toLowerCase());
+    }
+
+    function _getAllColors(editor) {
```

Comment:
this pass takes around 2ms and is being done already in the color hints extension, we should isolate this to another core lib for code intelligence to remove redundant checks that are computationally intensive. The plan was to create a core code intelligence as part of this project: https://github.com/orgs/phcode-dev/projects/13

If we do intensive computation on every type, the performance wil go down a lot, so we need to optimize it